src/Gateways/AbstractGateway.php

```diff
@@ -831,6 +900,10 @@ public function process_refund( $order_id, $amount = null, $reason = '' ) {
 
 		$order        = new WC_Order( $order_id );
 		$request      = $this->prepare_request( $txn_type, $order );
+		$request->set_request_data( array(
+			'refund_amount' => $amount,
+			'refund_reason' => $reason,
+		));
 		$request_args = $request->get_args();
 		if ( 0 >= (float)$request_args[ RequestArg::AMOUNT ] ) {
```

Comment:
We should validate the refund amount. You can check https://github.heartlandapps.net/DevPortal/opencart-hybrid/blob/f29fe16495ce0bf62c70b94d94242d6aa6555f9a/upload/admin/controller/extension/payment/globalpayments_ucp.php#L411

Reply:
I tested, and if we pass a refund value > refundable order value, there will be an error the same as in the admin.

{ ["error"]=> array(1) { [0]=> string(22) "Invalid refund amount." } }   is resolved.

Reply:
@javiercrac I still consider that we need to do our own validation.
e.g. an amount of "0,12" will create a refund of "12"